### PR TITLE
Bugfix: 3075

### PR DESCRIPTION
**Pull request by Xbar**
1. Fix LaRank freeze Issue: https://github.com/shogun-toolbox/shogun/issues/3075
2. Minor style changes to LaRank unit test

### CODE REVIEW

File: src/shogun/multiclass/LaRank.h [hunk 1]
```diff
@@ -485,6 +485,8 @@ namespace shogun
 
 			/// progess output
 			int32_t step;
+                        
+                        static const int32_t MAX_ITERATION = 1000;
```
Comment (vigsterkr) [inline src/shogun/multiclass/LaRank.h]: mmm why a static, why not a settable property of LaRank?

File: tests/unit/multiclass/LaRank_unittest.cc [hunk 2]
```diff
@@ -6,56 +6,54 @@
...
 using namespace shogun;
 
-TEST(DISABLED_LaRank,train)
+TEST(LaRank,train)
 {
-	index_t num_vec=10;
-	index_t num_feat=3;
-	index_t num_class=num_feat; // to make data easy
-	float64_t distance=15;
+	index_t num_vec = 10;
```
Comment (vigsterkr) [inline tests/unit/multiclass/LaRank_unittest.cc]: please refrain from whitespace refactorings (see this whole unittest patch)... it just makes git bisecting more complicated. just make sure that your patches are following the codestyle requirements

File: src/shogun/multiclass/LaRank.cpp [hunk 3]
```diff
@@ -653,16 +653,20 @@ bool CLaRank::train_machine(CFeatures* data)
...
 		SG_ABS_PROGRESS(gap, -CMath::log10(gap), -CMath::log10(DBL_MAX), -CMath::log10(get_C()), 6)
 
 		if (!batch_mode)        // skip stopping criteria if online mode
 			gap = 0;
+                n_it++;
 	}
 	SG_DONE()
 
+        if (n_it >= MAX_ITERATION && gap > get_C())
+            SG_WARNING("LaRank exceeded max iteration but not converged.")
```
Comment (karlnapf) [inline src/shogun/multiclass/LaRank.cpp]: La Rank did not converge after the maximum number of %d iterations

File: src/shogun/multiclass/LaRank.h (hunk 1 above)
Comment (karlnapf) [inline src/shogun/multiclass/LaRank.h]: definitely

Comment (karlnapf) [inline src/shogun/multiclass/LaRank.h]: Also, something more well informed. How many iterations does this usually take until convergence? What does the literature say here?

File: tests/unit/multiclass/LaRank_unittest.cc (hunk 2 above)
Comment (karlnapf) [inline tests/unit/multiclass/LaRank_unittest.cc]: the best thing is to do a first patch with the bugfix, and a second with the whitespace changes

Comment (karlnapf) [inline tests/unit/multiclass/LaRank_unittest.cc]: Currently it is not clear what actually changed

**Comment by karlnapf** [comment]
Great! Thanks for the patch.
Can you address our minor comments? I think it can be merged then....travis is fine

**Comment by Xbar** [comment]
Changed max_iteration to be an option
Reverted most changes to the unit test

Max iteration is hard to set, because morbid cases can always be made that will take very long to converge. In practice, if iteration exceeds a fairly large number, it may reflect some problems in the training set, or how SVM is initialized.

File: src/shogun/multiclass/LaRank.h [hunk 4]
```diff
@@ -378,6 +378,17 @@ namespace shogun
 			 * @return tau
 			 */
 			float64_t get_tau() { return tau; };
+                        /** set max iterations
```
Comment (vigsterkr) [inline src/shogun/multiclass/LaRank.h]: indentation on this is wrong

File: src/shogun/multiclass/LaRank.h [hunk 5]
```diff
@@ -485,6 +496,8 @@ namespace shogun
 
 			/// progess output
 			int32_t step;
+                        
+                        int32_t max_iteration;
```
Comment (vigsterkr) [inline src/shogun/multiclass/LaRank.h]: same here

**Comment by vigsterkr** [comment]
@Xbar what all these unrelated doxygen changes? i mean it's great but it shoudl be part of separate PR, not about fixing LaRank

**Comment by Xbar** [comment]
I was trying to do a rebase, but I didn't do it the right way. Somehow it merged other commits from other people.

**Comment by Xbar** [comment]
@vigsterkr Could you close this PR?
Sorry but I really made a mess in my forked repo after an improper rebase.
I deleted my repo and created a new PR.

**Pull request closed without merging**